coreos-virt-install

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3
+# Wrapper for virt-install designed to run in a container
+# environment, and using defaults targeted for CoreOS.
+# https://github.com/cgwalters/coreos-assembler/issues/7
+# Today, this requires a Kickstart (and an installer ISO path);
+# we might change things down the line to have defaults for
+# one or both.
+
+import os,sys,argparse,subprocess,io,time,re
+
+def fatal(msg):
+    print('error: ' + msg, file=sys.stderr)
+    sys.exit(1)
+
+if os.getuid() == 0:
+    fatal("This script should be run as non-root in a container")
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dest", help="Destination disk",
+                    action='store', required=True)
+parser.add_argument("--create-disk", help="Automatically create disk as qcow2, parsing kickstart for size",
+                    action='store_true')
+parser.add_argument("--kickstart", help="Kickstart path",
+                    action='store', required=True)
+parser.add_argument("--location", help="Installer location",
+                    action='store', required=True)
+parser.add_argument("--memory", help="Memory size in MB",
+                    action='store', type=int, default=2048)
+parser.add_argument("--console-log-file", help="Console log file",
+                    action='store')
+parser.add_argument("--wait", help="Number of minutes to wait",
+                    action='store', type=int, default=10)
+args = parser.parse_args()
+
+# Copy of bits from https://pagure.io/standard-test-roles/pull-request/223
+def get_libvirt_smp_arg():
+    """Determine the number of CPUs that should be visible in the guest.
+    See e.g. https://www.redhat.com/archives/libvirt-users/2017-June/msg00025.html
+    We want to match the number of host physical cores.
+    """
+    # We may be run in a cgroup with fewer cores available than physical.
+    available_cpu = int(subprocess.check_output(['nproc']).strip())
```

Comment:
nit: This call could be simplified using a stdlib such as `multiprocessing`

```python
from multiprocessing import cpu_count

available_cpu = cpu_count()
```

Verification:
```
$ python3 -c "import multiprocessing; print(multiprocessing.cpu_count())"
4
$ nproc 
4
```

Reply:
It's intentional; `nproc` is smarter.

```
# podman run --cpuset-cpus 0 --rm -ti registry.fedoraproject.org/fedora:28 nproc
1
# podman run --cpuset-cpus 0 --rm -ti registry.fedoraproject.org/fedora:28 python3 -c 'import multiprocessing; print(multiprocessing.cpu_count())'
8
#
```